grpc/cmd/server/main.go

```diff
@@ -29,11 +29,26 @@ func main() {
 	}
 	defer lis.Close()
 
-	go runGRPC(lis)
-	runHTTP(clientAddr)
+	//this channel will be used to halt
```

Comment:
I just thought it would be better if you would have one place to catch your errors. And using an error channel is good way to do that instead of catching it on different places.